src/Illuminate/Support/Collection.php

```diff
@@ -183,6 +183,42 @@ public function contains($key, $operator = null, $value = null)
         return $this->contains($this->operatorForWhere(...func_get_args()));
     }
 
+    /**
+     * Determine if all items exist in the collection.
+     *
+     * @param  mixed $needles
+     *
+     * @return bool
+     */
+    public function containsAll($needles)
+    {
+        foreach ($this->getArrayableItems($needles) as $needle) {
+            if (! $this->contains($needle)) {
+                return false;
+            }
+        }
+
+        return true;
+    }
```

Comment:
This is O(NM).

This should either use a [`Set`](https://www.php.net/manual/en/class.ds-set.php), or just compare it with a `diff`:

```php
return new static($needles)->diff($this)->isEmpty();
```

Reply:
Nice, this takes away the need for the strict comparison as well.

Reply:
unfortunately `array_diff()` won't work with multi dimensional arrays.

> Two elements are considered equal if and only if (string) $elem1 === (string) $elem2. In other words: when the string representation is the same. 

PHP 7.3 and lower throw an error.